packages/x-date-pickers-pro/src/PickersRangeCalendarHeader/PickersRangeCalendarHeader.tsx

```diff
@@ -0,0 +1,160 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { styled, useThemeProps } from '@mui/material/styles';
+import { PickersCalendarHeader } from '@mui/x-date-pickers/PickersCalendarHeader';
+import { PickerValidDate } from '@mui/x-date-pickers/models';
+import {
+  PickersArrowSwitcher,
+  useLocaleText,
+  useNextMonthDisabled,
+  usePreviousMonthDisabled,
+  useUtils,
+} from '@mui/x-date-pickers/internals';
+import { PickersRangeCalendarHeaderProps } from './PickersRangeCalendarHeader.types';
+
+type PickersRangeCalendarHeaderComponent = (<TDate extends PickerValidDate>(
+  props: PickersRangeCalendarHeaderProps<TDate> & React.RefAttributes<HTMLDivElement>,
+) => React.JSX.Element) & { propTypes?: any };
+
+const PickersRangeCalendarHeaderContentSingleCalendar = styled(PickersCalendarHeader, {
+  name: 'PickersRangeCalendarHeader',
+  slot: 'ContentSingleCalendar',
+  overridesResolver: (_, styles) => styles.contentSingleCalendar,
+})({}) as typeof PickersCalendarHeader;
+
+const PickersRangeCalendarHeaderContentMultipleCalendars = styled(PickersArrowSwitcher, {
+  name: 'PickersRangeCalendarHeader',
+  slot: 'ContentMultipleCalendars',
+  overridesResolver: (_, styles) => styles.contentMultipleCalendars,
+})({
+  padding: '12px 16px 4px 16px',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'space-between',
+});
+
+const PickersRangeCalendarHeader = React.forwardRef(function PickersRangeCalendarHeader<
+  TDate extends PickerValidDate,
+>(inProps: PickersRangeCalendarHeaderProps<TDate>, ref: React.Ref<HTMLDivElement>) {
+  const props = useThemeProps({ props: inProps, name: 'MuiPickersRangeCalendarHeader' });
```

Comment:
Do you think it make sense to register the `PickersRangeCalendarHeader` component inside the theme like I did or not?

Reply:
IMHO, in its current form - it doesn't make much sense. 🤔 
The props are the same as for the `PickersCalendarHeader`, the styled component can't be overridden via `styleOverrides` and there are no explicit `classes` for this component. 🤔 


I feel that without "dog-fooding" this implementation we won't know what is actually missing.
I would be in favor of introducing 1 or 2 new doc examples of overriding `pickersCalendarHeader` on `DateRangePicker` with a custom component as well as trying to style only the range calendar header.
WDYT?